### PR TITLE
Fixing CopyTextureRegion going out of bounds when src_box is null.

### PR DESCRIPTION
**Pull request by dave-juicelabs**

### CODE REVIEW

**Comment by dave-juicelabs** [comment]
This issue came about by having a src image with size (1840, 1840, 1) and a destination image with size (1024, 1024, 1). src_box was null and the src image size was causing an error during vkCmdCopyImage.

File: libs/vkd3d/command.c [hunk 1]
```diff
@@ -5362,8 +5362,22 @@ static void vk_image_copy_from_d3d12(VkImageCopy *image_copy,
...
     else
     {
+        VkExtent3D dstExtent;
+
+        // Calculate the extent based on the source first
         unsigned int miplevel = image_copy->srcSubresource.mipLevel;
         vk_extent_3d_from_d3d12_miplevel(&image_copy->extent, src_desc, miplevel);
+
+        // Now make sure it does not extend past the destination when there is no src_box
+        miplevel = image_copy->dstSubresource.mipLevel;
```
Comment (Joshua-Ashton) [inline libs/vkd3d/command.c:5372]: Just pass this directly to vk_extent_3d_from_d3d12_miplevel... Same for the above srcSubresource.

File: libs/vkd3d/command.c [hunk 2]
```diff
@@ -5362,8 +5362,22 @@ static void vk_image_copy_from_d3d12(VkImageCopy *image_copy,
...
+        vk_extent_3d_from_d3d12_miplevel(&dstExtent, dst_desc, miplevel);
+
+        if((dst_x + image_copy->extent.width) > dstExtent.width)
```
Comment (Joshua-Ashton) [inline libs/vkd3d/command.c:5375]: Use min here.

**Review by Joshua-Ashton — CHANGES_REQUESTED** [review]
Please add a test case for this also.

**Review by Joshua-Ashton — CHANGES_REQUESTED** [review]
Please split the test change into another commit before the original one.

**Review by HansKristian-Work — APPROVED** [review]
LGTM

**Pull request merged**